### DISCUSSION TITLE
Missing function to convert numeric values to string or autoconversion

### DISCUSSION THREAD

**Discussion by co-xpts** [datalust/seq-tickets#1776] [Help] [answered]
To have a nice bar chart based on the combination of two values I tried:

(text: string, num: numeric)
`select count(*) from stream group by Concat(text, ' ', num)`
which results in a single group with a title of "null" as the concat fails to concatenate, I guess.

Is there an explicit conversion from numeric to string?
Or even better, can you autoconvert numeric values when used in as a parameter which expects a string?

**Comment by liammclennan** [comment, marked as answer]
Hi @co-bbrzgruppe,

[`ToJson`](https://docs.datalust.co/docs/scalar-functions#tojsonarg) will convert values to strings. Try `select count(*) from stream group by Concat(text, ' ', ToJson(num))`

**Reply by co-xpts** [reply]
Aaah! Confirming it works. But it's far from intuitive.
Maybe add a comment and an example to https://docs.datalust.co/docs/scalar-functions ?

**Reply by liammclennan** [reply]
Good idea. I will add something to the docs. Thanks.
